src/mono/wasm/Wasm.Build.Tests/ProjectProviderBase.cs

```diff
@@ -380,6 +381,8 @@ public void AssertIcuAssets(AssertBundleOptionsBase assertOptions)
         }
 
         IEnumerable<string> actual = Directory.EnumerateFiles(assertOptions.BinFrameworkDir, "icudt*dat");
+        if (assertOptions.GlobalizationMode == GlobalizationMode.Hybrid)
```

Comment:
Just curious, why `actual` is not all files ? Like `*.*` ? @radical

Reply:
because are only looking for the icu related files here, and we do throw if the number of expected!=actual .

Reply:
But now that we have a differently named file too, it would indeed make sense to enumerate all the files ignoring subdirectories, and `AssertFilesOnDisk` can skip the number of files check.

Reply:
@radical @pavelsavara Should I make the change as part of this PR? If I understand @radical comment correctly, it would require removing the number of files check and replacing `Assert.Equal(expected, actualFileNames)` with `Assert.Contains(expected, actualFileNames)` in the `AssertFilesOnDisk` function.

Reply:
Follow up PR would be fine.